src/ai/backend/storage/filebrowser/filebrowser.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import annotations
+
+import logging
+from contextlib import asynccontextmanager
+from datetime import datetime
+from pathlib import Path
+from typing import Any, List, Mapping, Type
+from uuid import UUID
+
+import aiodocker
+import pkg_resources
+
+from ai.backend.common.logging import BraceStyleAdapter
+from ai.backend.common.validators import BinarySize
+from ai.backend.storage.abc import AbstractVolume
+from ai.backend.storage.context import Context
+from ai.backend.storage.utils import get_available_port
+from ai.backend.storage.vfs import BaseVolume
+
+from .config_browser_app import prepare_filebrowser_app_config
+from .database import FilebrowserTrackerDB
+
+BACKENDS: Mapping[str, Type[AbstractVolume]] = {
+    "vfs": BaseVolume,
+}
+
+
+log = BraceStyleAdapter(logging.getLogger(__name__))
+
+__all__ = (
+    "create_or_update",
+    "destroy_container",
+    "get_container_by_id",
+    "get_filebrowsers",
+    "get_network_stats",
+)
+
+
+@asynccontextmanager
+async def closing_async(thing):
+    try:
+        yield thing
+    finally:
+        await thing.close()
+
+
+async def create_or_update(
+    ctx: Context,
+    host: str,
+    vfolders: list[dict],
+) -> tuple[str, int, str]:
+    image = ctx.local_config["filebrowser"]["image"]
+    service_ip = ctx.local_config["filebrowser"]["service_ip"]
+    service_port = ctx.local_config["filebrowser"]["service_port"]
+    max_containers = ctx.local_config["filebrowser"]["max_containers"]
+    user_id = ctx.local_config["filebrowser"]["user_id"]
+    group_id = ctx.local_config["filebrowser"]["group_id"]
+    cpu_count = ctx.local_config["filebrowser"]["max_cpu"]
+    memory = ctx.local_config["filebrowser"]["max_mem"]
+    memory = int(BinarySize().check_and_return(memory))
```

Comment:
`filebrowser.max_mem` is already declared as `tx.BinarySize` on trafaret scheme initiation, so I think we can just skip this step.